### PR TITLE
fix(coredump): remove bash core generated by scylla_setup

### LINKED ISSUE #2015: an expected coredump in scylla_setup causes artifact-test fails
### Prerequisites

* [x] Are you rebased to master ?
* [ ] Is it reproducible ?
    - only found once.
* [x] Did you perform a cursory search if this issue isn't opened ?

### Versions
* **SCT**: branch master
* **scylla**:  666.development-0.20200407.16246d1c99e
* Cluster size: 1
* OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-0d6a24fe35fdf5dc4 (eu-west-1)

### Logs
* test_id: 4c259718-60ce-4621-ba4a-c99f4369a314
* job log: sct log: https://cloudius-jenkins-test.s3.amazonaws.com/4c259718-60ce-4621-ba4a-c99f4369a314/20200408_001125/sct-runner-4c259718.zip
* db log: https://cloudius-jenkins-test.s3.amazonaws.com/4c259718-60ce-4621-ba4a-c99f4369a314/20200408_001125/db-cluster-4c259718.zip

### Description

This test failed for a coredump in scylla_setup (coredump setup), the coredump is generated for 'bash' process as expected.We need to ignore the special coredump in SCT.

### Steps to Reproduce

1. run artifact-test with oel

**Expected behavior:** [What you expected to happen]

The coredump should be ignored by SCT, because it's expected.

**Actual behavior:** [What actually happened]

Coredump occurred during scylla_setup, it's generated for 'bash' process.

coredump file: https://storage.cloud.google.com/upload.scylladb.com/core.bash.0.f0b1893e3c264f899ee31dee60608402.12356.1586304086000000/core.bash.0.f0b1893e3c264f899ee31dee60608402.12356.1586304086000000.gz

```
2020-04-08 00:01:37.505: (CoreDumpEvent Severity.ERROR): node=Node artifacts-oel76-jenkins-db-node-4c259718-1 [54.194.6.193 | 10.0.3.227] (seed: True)
corefile_url=
https://storage.cloud.google.com/upload.scylladb.com/core.bash.0.f0b1893e3c264f899ee31dee60608402.12356.1586304086000000/core.bash.0.f0b1893e3c264f899ee31dee60608402.12356.1586304086000000.gz
backtrace=           PID: 12356 (bash)
           UID: 0 (root)
           GID: 0 (root)
        Signal: 11 (SEGV)
     Timestamp: Wed 2020-04-08 01:01:26 BST (5s ago)
  Command Line: /bin/bash /tmp/tmpekbbc8hl
    Executable: /usr/bin/bash
 Control Group: /user.slice/user-1000.slice/session-2.scope
          Unit: session-2.scope
         Slice: user-1000.slice
       Session: 2
     Owner UID: 1000 (clckwrk)
       Boot ID: f0b1893e3c264f899ee31dee60608402
    Machine ID: 677cfa1df55d442ea4279d4877a617e3
      Hostname: artifacts-oel76-jenkins-db-node-4c259718-1
      Coredump: /var/lib/systemd/coredump/core.bash.0.f0b1893e3c264f899ee31dee60608402.12356.1586304086000000
       Message: Process 12356 (bash) of user 0 dumped core.
                
                Stack trace of thread 12356:
                #0  0x00007fe5b08c7597 kill (libc.so.6)
                #1  0x0000000000440686 kill_pid (bash)
                #2  0x0000000000472dfe kill_builtin (bash)
                #3  0x000000000042f23f execute_builtin.isra.2 (bash)
                #4  0x0000000000431329 execute_simple_command (bash)
                #5  0x0000000000432233 execute_command_internal (bash)
                #6  0x0000000000433c1e execute_command (bash)
                #7  0x000000000041e255 reader_loop (bash)
                #8  0x000000000041c8be main (bash)
                #9  0x00007fe5b08b3495 __libc_start_main (libc.so.6)
                #10 0x000000000041d35a _start (bash)

download_instructions=
gsutil cp gs://upload.scylladb.com/core.bash.0.f0b1893e3c264f899ee31dee60608402.12356.1586304086000000/core.bash.0.f0b1893e3c264f899ee31dee60608402.12356.1586304086000000.gz .
gunzip /var/lib/systemd/coredump/core.bash.0.f0b1893e3c264f899ee31dee60608402.12356.1586304086000000.gz
```

### PR DESCRIPTION
**Pull request by enaydanov**
Trello: https://trello.com/c/Y9Tbnp6D/1836-ignore-coredump-that-is-created-during-scyllasetup

Closes #2015 

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2016,23 +2015,28 @@ def scylla_setup(self, disks):
...
+                for conf in ['io.conf', 'io_properties.yaml']:
+                    self.remoter.send_files(src=os.path.join('./configurations/', conf),  # pylint: disable=not-callable
+                                            dst='/tmp/')
+                    self.remoter.run('sudo mv /tmp/{0} /etc/scylla.d/{0}'.format(conf))
+            else:
+                self.remoter.run('sudo /usr/lib/scylla/scylla_setup --nic {} --disks {} {}'
+                                 .format(devname, ','.join(disks), extra_setup_args))
+
+            # Remove bash core which generated by scylla_setup for internal test purpose.
+            self.remoter.run("sudo rm /var/lib/systemd/coredump/core.bash.*")
```
Comment (bentsi) [inline sdcm/cluster.py:2039]: add `ignore_status=True` since the core may not be there...

Comment (enaydanov) [reply sdcm/cluster.py]: done

**Comment by roydahan** [comment]
The solution is unacceptable.

The whole idea behind this coredump is to verify that there is no issue with creating coredumps.

We need the artifact test to verify that if the setup initiated this coredump, it exist in the expected location.

**Comment by bentsi** [comment]
@roydahan we delete the coredump after scylla_setup finished and checked that the coredump mechanism works...
There is a bug(
https://github.com/scylladb/scylla/issues/6159) in scylla_coredump_setup that doesn't clean the "test coredump" after it finished

**Comment by bentsi** [comment]
following commit introduced the bug: https://github.com/scylladb/scylla/commit/5f18964763608b5e5d5a833c0777e253d2c21624

**Comment by roydahan** [comment]
So, Scylla_setup is responsible on testing the coredump?
If it fails to create the coredump, does it fail the setup?

**Comment by enaydanov** [comment]
> So, Scylla_setup is responsible on testing the coredump?
> If it fails to create the coredump, does it fail the setup?

Yes. There is a code:

```python
...
        try:
            run('coredumpctl --no-pager --no-legend info {}'.format(pid))
        except subprocess.CalledProcessError as e:
            print('Does not able to detect coredump, failed to configure systemd-coredump.')
            sys.exit(1)
...
```

**Pull request merged**